### PR TITLE
Fix clip rect for iframes in hit testing code

### LINKED ISSUE #8080: iframe hit test is wrong
In the following example, the content of the iframe turns blue on `:hover`. But moving the mouse at the top or the bottom of the iframe, the content of the iframe is considered not `:hover` (so staying red). I guess it's a hit test issue. This makes clicking on links impossible in a more complex layout.

a.html:

```html
<style>
  body {
    margin: 0;
  }

  div {
    width: 10px;
    height: 160px;
    background: green;
  }

  iframe {
    display: block;
    width: 200px;
    height: 200px;
    border: 0;
  }

</style>

<div></div>

<iframe src="b.html" mozbrowser="true"></iframe>
```

b.html:
```html
<style>
  html { background: red }
  html:hover { background: blue }
</style>
```

### PR DESCRIPTION
**Pull request by paulrouget**
Fixes #8080 r? @mrobinson 

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8671)

### CODE REVIEW

**Comment by paulrouget** [comment]
This is a rebase of https://github.com/servo/servo/pull/8557

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -1934,45 +1936,53 @@ impl<Window: WindowMethods> IOCompositor<Window> {
...
         let layer_bounds = *layer.bounds.borrow();
         let masks_to_bounds = *layer.masks_to_bounds.borrow();
         if layer_bounds.is_empty() && masks_to_bounds {
             return None;
         }
+        let scroll_offset = layer.extra_data.borrow().scroll_offset;
 
-        let clipped_layer_bounds = match clip_rect.intersection(&layer_bounds) {
+        // Total offset from parent coordinates to this layer's coordinates.
+        // FIXME: This offset is incorrect for fixed-position layers.
```
Comment (pcwalton) [inline components/compositing/compositor.rs]: Could we get an issue on file for this? It shouldn't be hard to fix.

**Comment by pcwalton** [comment]
This looks good to me.

**Comment by paulrouget** [comment]
https://github.com/servo/servo/issues/8672

File: components/compositing/compositor.rs (hunk 1 above)
Comment (paulrouget) [inline components/compositing/compositor.rs]: done: https://github.com/servo/servo/issues/8672

**Comment by glennw** [comment]
r=pcwalton

**Comment by paulrouget** [comment]
```

Tests with unexpected results:
  ▶ TIMEOUT [expected FAIL] /css-flexbox-1_dev/html/flexbox-basic-iframe-vert-001.htm
  │ 
  │ was ready to save, resetting ready_to_save_state
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ was ready to save, resetting ready_to_save_state
  └ Shutting down the Constellation after generating an output file or exit flag specified

  ▶ TIMEOUT [expected FAIL] /css-flexbox-1_dev/html/flexbox-basic-iframe-horiz-001.htm
  │ 
  │ was ready to save, resetting ready_to_save_state
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ was ready to save, resetting ready_to_save_state
  └ Shutting down the Constellation after generating an output file or exit flag specified
```

I can't reproduce locally (osx):

```
./mach test-css --release  tests/wpt/css-tests/css-flexbox-1_dev/html/flexbox-basic-iframe-horiz-001.htm
```

Not sure what to do next

**Comment by jdm** [comment]
@bors-servo: retry
* #8676

**Comment by paulrouget** [comment]
Looks like a different issue this time?

```
  ▶ TIMEOUT [expected PASS] /_mozilla/css/canvas_over_area.html
  │ 
  │ updating composition_request (NewFrameTree)
  │ not ready to composite: NotReadyToPaintImage(JustNotifiedConstellation)
  │ not ready to composite: NotReadyToPaintImage(WaitingOnConstellation)
  │ not ready to composite: NotReadyToPaintImage(WaitingOnConstellation)
  │ not ready to composite: NotReadyToPaintImage(WaitingOnConstellation)
  │ not ready to composite: NotReadyToPaintImage(WaitingOnConstellation)
[…]
```

**Comment by jdm** [comment]
@bors-servo: retry
* #8311

**Comment by jdm** [comment]
@bors-servo: retry
* sorry, playing with builders interfered

**Comment by jdm** [comment]
@bors-servo: retry
* more builder investgiations

**Comment by jdm** [comment]
@bors-servo: retry
* #8676

**Pull request merged**